route_info/src/lib.rs

```diff
@@ -0,0 +1,76 @@
+use geom::{Distance, FindClosest, LonLat};
+use map_model::{IntersectionID, Map};
+use wasm_bindgen::prelude::*;
+
+#[wasm_bindgen]
+pub struct RouteInfo {
+    map: Map,
+    closest_intersection: FindClosest<IntersectionID>,
+}
+
+#[wasm_bindgen]
+impl RouteInfo {
+    #[wasm_bindgen(constructor)]
+    pub fn new(map_bytes: &[u8]) -> Result<RouteInfo, JsValue> {
+        // Panics shouldn't happen, but if they do, console.log them.
+        console_error_panic_hook::set_once();
+
+        web_sys::console::log_1(&format!("Got {} bytes, deserializing", map_bytes.len()).into());
+
+        let map: Map = bincode::deserialize(map_bytes).map_err(err_to_js)?;
```

Comment:
So after we pass in a binary map file (produced in abstreet-to-atip and soon hosted in S3), we deserialize it. Then we also build a quadtree to snap to an intersection, since all the queries will make use of that